### PR TITLE
fix: cleans up employees that are added by mistake

### PR DESCRIPTION
**Pull request by mikaelbr**
In cases where bemanning isn't correctly used, the start date defaults to 2018 causing some employees to be added where they shouldn't be. These are never updated as we update only employees with a start date.

We might update everyone, but that also means we add every employee we don't have to add. This simply removes them when rerunning orchestration.

Risk here is that we also remove potentially added metadata (address, contact etc) but I guess they shouldn't have existed in the first place so it is OK.

### CODE REVIEW

File: src/Infrastructure/OrchestratorService.cs [hunk 1]
```diff
@@ -46,6 +46,17 @@ public async Task FetchMapAndSaveEmployeeData()
...
+        {
+            // Remove potential employees that shouldn't have been added.
+            // This should normally not happen, but might happen in cases where
+            // StartDate in bemanning wasn't set properly when orchestrating.
+            _logger.LogInformation(
+                "Deleting employee with email {BemanningEmail} from database, since they haven't started yet",
+                bemanning.Email);
+            await _employeesRepository.EnsureEmployeeIsDeleted(bemanning.Email);
+        }
+
```
Comment (haakoaho) [inline src/Infrastructure/OrchestratorService.cs:59]: Skal denne koden kjøres bare én gang? Isåfall må den jo ryddes opp i i etterkant.

File: src/Infrastructure/OrchestratorService.cs [hunk 2]
```diff
@@ -46,6 +46,17 @@ public async Task FetchMapAndSaveEmployeeData()
...
 
+        foreach (var bemanning in bemanningEntries.Where(employee => !IsActiveEmployee(employee)))
+        {
+            // Remove potential employees that shouldn't have been added.
+            // This should normally not happen, but might happen in cases where
+            // StartDate in bemanning wasn't set properly when orchestrating.
+            _logger.LogInformation(
+                "Deleting employee with email {BemanningEmail} from database, since they haven't started yet",
+                bemanning.Email);
+            await _employeesRepository.EnsureEmployeeIsDeleted(bemanning.Email);
```
Comment (Yoeori) [inline src/Infrastructure/OrchestratorService.cs:57]: This possibly returns a blobUrl, at the end of the function a list of blobs of previous employees is cleaned. Could be nice to add returned images to that list.

File: src/Infrastructure/OrchestratorService.cs [hunk 3]
```diff
@@ -46,6 +46,17 @@ public async Task FetchMapAndSaveEmployeeData()
 
         var phoneNumberUtil = PhoneNumberUtil.GetInstance();
 
+        foreach (var bemanning in bemanningEntries.Where(employee => !IsActiveEmployee(employee)))
```
Comment (Yoeori) [inline src/Infrastructure/OrchestratorService.cs:49]: This loop also takes over the loop at the end of the function which removes employees with an endDate so the other one can be deleted.

**Review by Yoeori — CHANGES_REQUESTED** [review]

File: src/Infrastructure/OrchestratorService.cs (hunk 1 above)
Comment (mikaelbr) [reply src/Infrastructure/OrchestratorService.cs:59]: Nei, egentlig ikke. Det kan fort skje i periodisk kjøring av orkestrering. Det kan skje at du legger til en ny ansatt i bemanning som ikke får satt prosjektet `Ikke startet` før chewbacca-orkestrering kjører. Da vil du komme i tilsvarende tilstand. Så i de fleste tilfeller vil dette kjøre bare for å slette brukere som ikke lengre er aktive og slette det.

Teknisk sett kan det også kjøre ved sluttdato for enkelte og fjerne de helt. Det kan jo være uønsket, men også ønsket egentlig. Mulig vi også skulle ha slettet blob i så fall

File: src/Infrastructure/OrchestratorService.cs (hunk 3 above)
Comment (mikaelbr) [reply src/Infrastructure/OrchestratorService.cs:49]: That's a good point. Might actually just change that to have `EnsureInactiveEmployeesAreDeleted` 🤔 


The consequence though, is that employees before the start date won't be able to add contact information and allergies. That's not good either. But this is also true with existing implementation

Comment (mikaelbr) [reply src/Infrastructure/OrchestratorService.cs]: Refactored and fixed deleting blobs for both

File: src/Infrastructure/OrchestratorService.cs [hunk 4]
```diff
@@ -104,7 +93,21 @@ public async Task FetchMapAndSaveEmployeeData()
             }
         }
 
-        var blobUrlsToBeDeleted = await _employeesRepository.EnsureEmployeesWithEndDateBeforeTodayAreDeleted();
+        // Remove employees with end date that has been passed.
+        var blobUrlsToBeDeleted = (await _employeesRepository.EnsureEmployeesWithEndDateBeforeTodayAreDeleted()).ToList();
+
+        foreach (var bemanning in bemanningEntries.Where(employee => !IsActiveEmployee(employee)))
+        {
+            // Remove potential employees that shouldn't have been added.
```
Comment (Yoeori) [inline src/Infrastructure/OrchestratorService.cs:101]: And employees with endDate 😉

**Review by Yoeori — APPROVED** [review]

File: src/Infrastructure/OrchestratorService.cs (hunk 4 above)
Comment (mikaelbr) [reply src/Infrastructure/OrchestratorService.cs:101]: Technically those would have been removed in the previous step and tried again (unnecessarily) to be removed here. Could potentially be explicit on IsBeforeStartDate

Comment (Yoeori) [reply src/Infrastructure/OrchestratorService.cs:101]: Oh yeah right, read over that line 😅

**Pull request merged**